File: src/Neuralm.Application.Messages/Requests/RegisterRequest.cs
namespace Neuralm.Application.Messages.Requests
{
    /// <summary>
    /// Represents the <see cref="RegisterRequest"/> class.
    /// </summary>
    public class RegisterRequest : Request
    {
        /// <summary>
        /// Gets the username.
        /// </summary>
        public string Username { get; }

        /// <summary>
        /// Gets the password.
        /// </summary>
        public string Password { get; }

        /// <summary>
        /// Gets the credential type code.
        /// </summary>
        public string CredentialTypeCode { get; }

        /// <summary>
        /// Initializes an instance of the <see cref="RegisterRequest"/> class.
        /// </summary>
        /// <param name="username">The username.</param>
        /// <param name="password">The password.</param>
        /// <param name="credentialTypeCode">The credential type code.</param>
        public RegisterRequest(string username, string password, string credentialTypeCode)
        {
            Username = username;
            Password = password;
            CredentialTypeCode = credentialTypeCode;
        }
    }
}

File: src/Neuralm.Presentation.Web/src/messages/requests/RegisterRequest.ts
import Request from './Request';

/**
 * Represents the RegisterRequest class.
 */
export default class RegisterRequest extends Request {
  public username: string;
  public password: string;
  public credentialTypeCode: string;

  /**
   * Initializes the RegisterRequest class.
   * @param username The user name.
   * @param password The password.
   */
  constructor(username: string, password: string) {
    super();
    this.username = username;
    this.password = password;
    this.credentialTypeCode = 'Name';
  }
}
